datafusion/core/tests/config.rs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion::common::extensions_options;
+
+// The structure is not really used. We just check that the macro re-export works and everything referenced within the
```

Comment:
Can you please move this code to https://github.com/apache/arrow-datafusion/tree/main/datafusion-examples/examples

That would also provide a good place maybe to provide some simple program showing how to use options created with this macro 



Maybe we could add a small test showing that the configuration could be instantiated and used 🤔  As a way to try and document its use.

Reply:
done